### ISSUE TITLE
pod install "No such file or directory" with branch from private repository

### ISSUE THREAD

**Issue by Dzamir** [CocoaPods/CocoaPods#3815]
I have a private pods repository with 2 pods and it works perfectly until I try to use a branch of a pod.

This works:

    pod 'AnotherPodInsidePrivateRepo', :head

But this doesn't

    pod 'AnotherPodInsidePrivateRepo', :git => 'git@ git.privaterepository.net:ios/avanatta-ios.git', :branch => 'refactor'

This is the error I get: 

### Command

```
/usr/bin/pod install
```

### Stack

```
   CocoaPods : 0.37.2
        Ruby : ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
    RubyGems : 2.0.14
        Host : Mac OS X 10.10.4 (14E46)
       Xcode : 6.4 (6E35b)
         Git : git version 2.3.2 (Apple Git-55)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : private-specs - git@git.privaterepository.net:ios/Specs.git @ 12f5750538023c3555341fee1c621d3ecd492e16
               master - https://github.com/CocoaPods/Specs.git @ d7e5796329ebdba8a740de1aaabfa5108f7f3752
```

### Plugins

```
cocoapods-plugins : 0.4.2
cocoapods-trunk   : 0.6.1
cocoapods-try     : 0.4.5
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
source 'git@git.privaterepository.net:ios/Specs.git'
platform :ios, '7.1'

pod 'PodInsidePrivateRepo', :head
pod 'AnotherPodInsidePrivateRepo', :git => 'git@ git.privaterepository.net:ios/avanatta-ios.git', :branch => 'refactor'
pod 'BPForms', '~> 2.0'
pod 'MBProgressHUD', '~> 0.9'
pod 'DZNWebViewController', '~> 2.0'
pod 'AFNetworking', '~> 2.0'
#pod 'AFAmazonS3Client', '~> 2.0'
pod 'AFNetworkActivityLogger', '~> 2.0'
pod 'NSDictionary+Accessors', '~> 1.0'
pod 'CCBottomRefreshControl', '0.3'
pod 'DateTools', '~> 1.4'
pod 'DZNEmptyDataSet', '~> 1.4'
pod 'HexColors', '~> 2.2'
pod 'GPUImage', '~> 0.1'
pod 'FXBlurView', '~> 1.6'
pod 'pop', '~> 1.0'
pod 'CHCircleGaugeView', '~> 1.1'
pod 'JazzHands', '~> 0.1'
pod 'VGParallaxHeader', '~> 0.0'
pod 'Hakawai', '~> 1.0'
pod 'UICKeyChainStore', '~> 1.0'
pod 'DAKeyboardControl', '2.4'
#pod 'ZFDragableModalTransition', '~> 0.2'
pod 'SDWebImage', :git => 'https://github.com/Dzamir/SDWebImage.git'
#pod 'STTweetLabel', '~> 3.1'
pod 'TOMSMorphingLabel', '~> 0.5'
pod 'PureLayout', '~> 2.0'
pod 'WMATweetView', '~> 0.0'
pod 'TransitionKit', '~> 2.1'
pod 'JDFTooltips', '~> 1.0'
pod 'APAddressBook'
pod 'TUSafariActivity', '~> 1.0'
#pod 'THContactPicker', '~> 0.0'
#pod 'KBContactsSelection', '1.2.3' #DO NOT UPDATE, EDITS WILL BE LOST
pod 'Google/Analytics', '~> 1.0.0'
pod 'ALAssetsLibrary-CustomPhotoAlbum', '~> 1.3'
pod 'DZATimer', '~> 0.1'
```

### Error

```
Errno::ENOENT - No such file or directory - /var/folders/mb/jxdyv2d91zncsmlwf8t7wqkh0000gp/T/d20150713-7047-vp6elu
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1552:in `stat'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1552:in `block in fu_each_src_dest'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1568:in `fu_each_src_dest0'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1550:in `fu_each_src_dest'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:434:in `cp_r'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/downloader.rb:42:in `block in download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:140:in `message'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/downloader.rb:40:in `download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/external_sources/abstract_external_source.rb:108:in `block in pre_download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:80:in `titled_section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/external_sources/abstract_external_source.rb:106:in `pre_download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/external_sources/downloader_source.rb:13:in `fetch'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:358:in `fetch_external_source'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:335:in `block (2 levels) in fetch_external_sources'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:334:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:334:in `block in fetch_external_sources'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:59:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:333:in `fetch_external_sources'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer/analyzer.rb:62:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:196:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:114:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/user_interface.rb:59:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:113:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/installer.rb:93:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command/project.rb:101:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.8.1/lib/claide/command.rb:312:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/lib/cocoapods/command.rb:46:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.37.2/bin/pod:44:in `<top (required)>'
/usr/bin/pod:23:in `load'
/usr/bin/pod:23:in `<main>'
```

**Comment by segiddins** [comment]
I believe this is fixed in the 0.38 beta, can you give that a try?

**Comment by Dzamir** [comment]
Tried with the 0.38 beta and it works perfectly, thanks
